### DISCUSSION TITLE
How to play music files?

### DISCUSSION THREAD

**Discussion by maxidev62** [CosmosOS/Cosmos#3053] [Q&A]
how to play wav audio files

**Comment by maxidev62** [comment]
also if I try to play a sound using the method from the documentation I get an error Expected a 'data' block

**Reply by ascpixi** [reply]
I'm guessing you're using the userkit, for which the WAV parser does not support any kind of metadata. There are two possible solutions:
- update to the dev kit, which is the recommended way to use Cosmos,
- remove any and all kind of metadata from the WAV file using a tool like Audacity.

**Comment by ascpixi** [comment]
You can follow the method outlined in the documentation here: https://cosmosos.github.io/articles/Kernel/Audio.html
It's recommended to use the devkit (i.e. a version of Cosmos manually built from the `master` branch) if you're going to be using WAV files with metadata.
